File: Basic/Reverse a string with spaces intact/reverse-a-string-with-spaces-intact.cpp
//{ Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution
{
  public:
  
        string reverseWithSpacesIntact (string s)
        {
            // for ( size_t l{}, r{ s.size()-1 }; l < r; )
            //     if      ( s[ l ] == ' ' ) ++l;
            //     else if ( s[ r ] == ' ' ) --r;
            //     else swap( s[ l++ ], s[ r-- ] );
            // return s;
            
            
            
            int len=s.length()-1;
            int i=0;
            while(i<len){
                
                if(s[i]==' ')
                 i++;
                else if(s[len]==' ')
                    len--;
                else {
                    swap(s[i],s[len]);
                
                i++;
                len--;
                }
            }
            return s;
        }
      

};

//{ Driver Code Starts.
int main()
{
    int t;
    cin >> t;
    cin.ignore ();
    while (t--)
    {
        string s;
        getline (cin, s);
        Solution ob;
        cout << ob.reverseWithSpacesIntact (s) << endl;
    }
    return 0;
}
// Contributed By: Pranay Bansal

// } Driver Code Ends

File: Basic/Reverse a string with spaces intact/reverse-a-string-with-spaces-intact.java
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine());
        while(t-- > 0)
        {
            String s;
            s = br.readLine();
            
            Solution ob = new Solution();
            
            System.out.println(ob.reverseWithSpacesIntact(s));    
        }
    }
}
// } Driver Code Ends


//User function Template for Java
class Solution
{
    String reverseWithSpacesIntact(String S)
    {
        // your code here
        String str = S.replace(" ", "");
        int count = str.length() - 1;
        
        StringBuilder sb = new StringBuilder();
        
        for(int i=0; i<S.length(); i++){
            if(S.charAt(i) == ' '){
                sb.append(' ');
            }
            else{
                sb.append(str.charAt(count--));
            }
        }
        
        return sb.toString();
    }
}